python/GafferUI/PathListingWidget.py

```diff
@@ -302,6 +303,8 @@ def setColumns( self, columns ) :
 			return
 
 		_GafferUI._pathListingWidgetSetColumns( GafferUI._qtAddress( self._qtWidget() ), columns )
+		self._qtWidget().setStretchLastColumn( not any( c.getResizeMode() == GafferUI.PathColumn.ResizeMode.Stretch for c in columns ) )
```

Comment:
It took me a minute to realise that we weren't controlling a Qt behaviour here. Since the value is always derived from the column modes, perhaps things are actually clearer if we omit the `[sg]etStretchLastColumn` methods and just let `_TreeView` determine what to do based on the columns directly? I think we're basically duplicating state at the moment, which makes it a bit harder to follow the logic. If we remove this, then I think that means `_TreeView` gets to encapsulate all the resize logic, rather than sharing it with PathListingWidget.

Reply:
Good point! I took a stab at this in a24f839.